docs/lab/05/index.md

```diff
@@ -0,0 +1,224 @@
+
+```rust
+#[embassy_executor::task]
+async fn task1(channel_sender: Sender<'static, ThreadModeRawMutex, bool, 64>) {
+    loop {
+        Timer::after_secs(1).await;
+        channel_sender.send(true).await;
+    }
+}
+```
+
+In the `main` function, we need to then wait for the signal, and once it's received, toggle the LED.
+
+```rust
+// ---- fn main() ----
+loop {
+    let value = CHANNEL.receive().await;
+    match value {
+        true => led.toggle().unwrap(),
+        false => info!("We got something else")
+    }
+}
+```
+
+:::info
+The reason we need all of this is because Rust doesn't allow us to mutably borrow more than once. To use a peripheral (say PWM) inside multiple tasks, 
+we would need to either move it inside the task entirely, or use a mutable reference to it. If we have multiple tasks, though, once we move our peripheral variable *inside* the first task,
+we can't pass it to another task, because the value was *moved* inside that task completely. And if we wanted to pass it as a mutable reference instead, we would quickly realize that Rust 
+doesn't allow multiple mutable references at once, to avoid concurrent modifications. So this is why we need to either declare a global, static `Mutex` that any task can access, to ensure 
+that the value cannot be modified concurrently by two different tasks, or use channels and keep the peripheral inside the `main` function.
+
+To better understand the concepts of ownership and borrowing in Rust, take a look at [chapter 4](https://doc.rust-lang.org/book/ch04-00-understanding-ownership.html) of the Rust Book.
+:::
+
+## Potentiometer
+
+A potentiometer is a three-terminal resistor with a sliding or rotating contact that forms an adjustable voltage divider. If only two terminals are used, one end and the wiper, it acts as a variable resistor or rheostat. A volume knob on a speaker is a potentiometer, for instance.
+
+![Potentiometer](images/potentiometer_pins.png)
+
+## Exercises
+
+1. Connect an LED to GP0, an RGB LED to GP1, GP2, GP4 and a potentiometer to ADC0. Use Kicad to draw the schematic. (**1p**)
+2. Change the monochromatic LED's intensity, using button A (SW_A) and button B(SW_B) on the Pico Explorer. Button A will increase the intensity, and button B will decrease it. (**2p**)
+
+:::tip
+- Use PWM to control the intensity.
+- Create two tasks, one for button A, one for button B. Use a channel to send commands from each button task to the main task.
+:::
+
+3. Control the RGB LED's color with buttons A, B, X and Y on the Pico Explorer. (**2p**)
+- Button A -> RGB = Red
+- Button B -> RGB = Green
+- Button X -> RGB = Blue
+- Button Y -> RGB = Led Off
+:::tip
+Use a separate task for each button. The RGB LED's color will be set in the main task.
+:::
+
+4. In addition to the four buttons, control the RGB LED's intensity with the potentiometer. (**3p**)
```

Comment:
Details this a little bit more, explaining what happens when a button is pressed. You can use mermaid flowcharts to explain the transitions.

Reply:
I added a flowchart, let me know if it looks readable.

Reply:
Works.